server/src/service.rs

```diff
@@ -28,6 +29,7 @@ pub struct Service {
     collections: Arc<Mutex<Vec<Collection>>>,
     // Other attributes
     manager: Arc<Mutex<ServiceManager>>,
+    prompt_index: Arc<RwLock<u32>>,
```

Comment:
Would put this in the ServiceManager instead and would pass that to newly created prompts as well